src/util/dsp/yule_walker.cpp

```diff
@@ -0,0 +1,98 @@
+//**************************************************************************
+//
+//	$file: yule_walker.c $
+//	$author: Martin Fouilleul $
+//	$date: 28/11/2018 $
+//	$description: a simple Yule-Walker system solver$
+//
+//**************************************************************************
+
+#include<string.h> // memset()
+
+namespace otto::dsp
+{
+//--------------------------------------------------------------------------
+//
+//	SolveYuleWalker() is a simple solver for Yule-Walker equations using a specialized version of the Levinson-Durbin algorithm.
+//
+//	arguments :
+//		- size : the size of the Toeplitz matrix and of result vector sigmaAndCoeffs
+//		- acov : a buffer containing the covariance coefficents for indices 0 to (size-1).
+//			     ie. [acov(0) acov(1) ... acov(n-1)] of length (size)
+//
+//		- sigmaAndCoeffs : result vector of length (size). The first element will be the variance of the signal, and the remaining elements will be the estimated coefficients of the autoregressive filter.
+//		- scratchBuffer : a block of memory used to hold intermediate computations. You must pass a buffer of size at least (2 * size * sizeof(float)).
+//
+//--------------------------------------------------------------------------
+
+void SolveYuleWalker(unsigned int size, const float* acov, float* sigmaAndCoeffs, char* scratchBuffer)
+{
+	//NOTE(martin): first we solve the system Mx = y, where M is the symmetrical Toeplitz matrix derived from the autocovariance acov and y = [1 0 ... 0]T
+	//		Then we extract sigma as 1/x[0] and the filter coefficients as coeff[k] = - sigma * x[k+1].
+	//		(note that the results are packed in sigmaAndCoeffs, so sigma = sigmaAndCoeff[0] and coeff[k] = sigmaAndCoeff[k+1] !)
+
+	//NOTE(martin): Clear and organize the scratch memory
+	memset(sigmaAndCoeffs, 0, size*sizeof(float));
+	memset(scratchBuffer, 0, 2*size*sizeof(float));
+
+	float* backwardPrev = (float*)scratchBuffer; scratchBuffer += size*sizeof(float);
+	float* backward = (float*)scratchBuffer; scratchBuffer += size*sizeof(float);
+
+	//NOTE(martin): Initialize our first induction variables
+	float inv = 1./acov[0];
+	sigmaAndCoeffs[0] = backwardPrev[size-1] = inv;
+
+	int lastB = size-1;
+
+	//NOTE(martin): Loop to compute our induction variables until we reach the appropriate vector size
+	for(unsigned int n = 2; n <= size; n++)
```

Comment:
```suggestion
	for(int n = 2; n <= acov.size(); n++)
```
Don't use unsigned for stuff like this